dashboard/modules/serve/sdk.py

```diff
@@ -0,0 +1,126 @@
+                "\nSent deploy request successfully!\n "
+                "* Use `serve status` to check your deployments' statuses.\n "
+                "* Use `serve info` to see your running Serve "
+                "application's configuration.\n"
+            )
+            cli_logger.newline()
+        else:
+            self._log_failed_request(response)
+
+    def get_info(self) -> Union[Dict, None]:
+        info_address = f"{self._address}/api/serve/deployments/"
+        response = requests.get(info_address)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            self._log_failed_request(response)
+
+    def get_status(self) -> Union[Dict, None]:
+        status_address = f"{self._address}/api/serve/deployments/status"
+        response = requests.get(status_address)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            self._log_failed_request(response)
+
+    def delete_application(self) -> None:
+        delete_address = f"{self._address}/api/serve/deployments/"
+        response = requests.delete(delete_address)
+        if response.status_code == 200:
+            cli_logger.newline()
+            cli_logger.success("\nSent delete request successfully!\n")
+            cli_logger.newline()
+        else:
+            self._log_failed_request(response)
+
+    def set_up_runtime_env(
+        self,
+        deployment: Deployment,
+        new_runtime_env: Optional[dict] = None,
+        new_working_dir: Optional[str] = None,
+    ) -> None:
+        """
+        1. If not None, new_runtime_env overwrites deployment's runtime_env.
+        2. Then, if not None, new_working_dir overwrites deployment's working_dir.
+        3. Then, uploads the deployment's working_dir if it's local.
+
+        Mutates the deployment.
+        """
+
+        runtime_env = {}
+
+        if deployment.ray_actor_options is not None:
+            runtime_env = deployment.ray_actor_options.get("runtime_env", {})
+        if new_runtime_env is not None:
+            runtime_env = new_runtime_env
+        if new_working_dir is not None:
+            runtime_env["working_dir"] = new_working_dir
+
+        self._upload_working_dir_if_needed(runtime_env)
+
+        if deployment.ray_actor_options is None:
+            deployment._ray_actor_options = {"runtime_env": runtime_env}
+        else:
+            deployment.ray_actor_options["runtime_env"] = runtime_env
+
+    def _log_failed_request(
+        self, response: requests.models.Response, address: str = None
+    ):
+        address = address or self._address
+        error_message = (
+            f"\nRequest to address {address} failed. Got response status code "
+            f"{response.status_code} with the following message:"
+            f"\n\n{response.text}"
+        )
+        cli_logger.newline()
+        cli_logger.error(error_message)
+        cli_logger.newline()
```

Comment:
This should raise an exception, no? Otherwise this call / script will exit with "success," which would be quite confusing.

Reply:
I replaced `_log_failed_request` with `self._raise_error`, which raises a `RuntimeError`.